synapse/storage/background_updates.py

```diff
@@ -628,7 +628,7 @@ def create_index_sqlite(conn: Connection) -> None:
                 logger.debug("[SQL] %s", sql)
                 c.execute(sql)
 
-        if isinstance(self.db_pool.engine, engines.PostgresEngine):
+        if isinstance(self.db_pool.engine, engines.postgres.PostgresEngine):
```

Comment:
Oh bugger. I forgot that we use `isinstance` everywhere, which means that `PostgresEngine` and `SqliteEngine` need to be importable everywhere.

Argh. Maybe my advice in #12962 was wrong.